salt/_modules/vault.py

```diff
@@ -0,0 +1,858 @@
+                        client.secrets.activedirectory.configure(
+                            **secret_engine.secret_config
+                        )
+                    if secret_engine.type == 'database':
+                        client.secrets.database.configure(
+                            **secret_engine.secret_config
+                        )
+
+                    log.info(
+                        'Finished provisioning specific configurations for "%s" secrets engine...', secret_engine.type)
+
+                if secret_engine.extra_config:
+                    log.info(
+                        'Provisioning extra conifgurations for for "%s" secrets engine...', secret_engine.type)
+
+                    if secret_engine.type == 'ad':
+                        # Get roles from vault
+                        existing_roles = None
+                        try:
+                            existing_roles = client.secrets.activedirectory.list_roles()
+                            log.debug(existing_roles)
+                        except Exception:
+                            raise
+
+                        # Add new roles
+                        local_roles = secret_engine.extra_config['roles']
+                        for key in local_roles:
+                            log.debug('AD Role ["%s"] -> Role %s',
+                                      str(key), local_roles[key])
+                            try:
+                                client.secrets.activedirectory.create_or_update_role(
+                                    name=key,
+                                    service_account_name=local_roles[key]['service_account_name'],
+                                    ttl=local_roles[key]['ttl']
+                                )
+                            except Exception:
+                                raise
+
+                        # Remove missing roles
+                        if existing_roles:
+                            for role in existing_roles:
+                                if role in {k.lower(): v for k, v in local_roles.items()}:
+                                    log.debug(
+                                        'AD role ["%s"] exists in configuration, no cleanup necessary', role)
+                                else:
+                                    log.debug(
+                                        'Ad role ["%s"] does not exists in configuration, deleting...', role)
+                                    client.secrets.activedirectory.delete_role(
+                                        name=role
+                                    )
+                                    log.debug(
+                                        'AD role has been ["%s"] deleted.', role)
+                else:
+                    log.debug(
+                        'Secret engine "%s" does not contain any extra configurations.', secret_engine.type
+                    )
+        except Exception:
```

Comment:
Since we are not actually handling any exception, just re-raising, this try/except block is not needed. Can remove it and let things raise exceptions normally.